src/tensorlake/function_executor/service.py

```diff
@@ -201,46 +281,67 @@ def initialize_invocation_state_server(
+
+        # Validate task request (input data)
+        request_validator = MessageValidator(task.request)
+        request_validator.required_serialized_object("function_input")
 
         # Fail with internal error as this happened due to wrong task routing to this Server.
         # If we run the wrongly routed task then it can steal data from this Server if it belongs
         # to a different customer.
-        if request.namespace != self._namespace:
+        if task.namespace != self._namespace:
             raise ValueError(
-                f"This Function Executor is not initialized for this namespace {request.namespace}"
+                f"This Function Executor is not initialized for this namespace {task.namespace}"
             )
-        if request.graph_name != self._graph_name:
+        if task.graph_name != self._graph_name:
             raise ValueError(
-                f"This Function Executor is not initialized for this graph_name {request.graph_name}"
+                f"This Function Executor is not initialized for this graph_name {task.graph_name}"
             )
-        if request.graph_version != self._graph_version:
+        if task.graph_version != self._graph_version:
             raise ValueError(
-                f"This Function Executor is not initialized for this graph_version {request.graph_version}"
+                f"This Function Executor is not initialized for this graph_version {task.graph_version}"
             )
-        if request.function_name != self._function_name:
+        if task.function_name != self._function_name:
             raise ValueError(
-                f"This Function Executor is not initialized for this function_name {request.function_name}"
+                f"This Function Executor is not initialized for this function_name {task.function_name}"
             )
 
-        return RunTaskHandler(
-            request=request,
-            invocation_state=ProxiedInvocationState(
-                request.task_id, self._invocation_state_proxy_server
-            ),
-            function_wrapper=self._function_wrapper,
-            function_stdout=self._function_stdout,
-            function_stderr=self._function_stderr,
-            graph_metadata=self._graph_metadata,
-            logger=self._logger,
-        ).run()
+    def run_task(
+        self, request: RunTaskRequest, context: grpc.ServicerContext
+    ) -> RunTaskResponse:
+        # Convert RunTaskRequest to Task for create_task
+        task = _run_task_request_to_task(request)
+        self.create_task(CreateTaskRequest(task=task), context)
+
+        try:
+            for response in self.await_task(
+                AwaitTaskRequest(task_id=request.task_id), context
+            ):
+                last_response = response
+        finally:
+            self.delete_task(DeleteTaskRequest(task_id=request.task_id), context)
+
+        assert last_response.WhichOneof("response") == "task_result"
```

Comment:
We'll remove this code soon but we never use Python asserts in production code. It's not necessary to use them and no one knows about them and how they work. We use error logs for unexpected situations.

Reply:
Noted - but yeah, we're removing this code anyway, this is just a quick way to throw an exception if the state is weird.